code/modules/mob/dead/new_player/poll.dm

```diff
@@ -513,14 +517,21 @@
 	if(!replytext)
 		to_chat(usr, "The text you entered was blank. Please correct the text and submit again.")
 		return
-	var/adminrank = sanitizeSQL(poll_check_voted(pollid, TRUE))
+	var/voted = poll_check_voted(pollid, TRUE)
+	if(isnull(voted))
+		return
+	var/adminrank = sanitizeSQL(poll_rank())
 	if(!adminrank)
 		return
 	replytext = sanitizeSQL(replytext)
 	if(!(length(replytext) > 0) || !(length(replytext) <= 8000))
 		to_chat(usr, "The text you entered was invalid or too long. Please correct the text and submit again.")
 		return
-	var/datum/DBQuery/query_text_vote = SSdbcore.NewQuery("INSERT INTO [format_table_name("poll_textreply")] (datetime ,pollid ,ckey ,ip ,replytext ,adminrank) VALUES (Now(), [pollid], '[ckey]', INET_ATON('[client.address]'), '[replytext]', '[adminrank]')")
+	var/datum/DBQuery/query_text_vote
+	if(!voted)
+		query_text_vote  = SSdbcore.NewQuery("INSERT INTO [format_table_name("poll_textreply")] (datetime ,pollid ,ckey ,ip ,replytext ,adminrank) VALUES (Now(), [pollid], '[ckey]', INET_ATON('[client.address]'), '[replytext]', '[adminrank]')")
+	else
+		query_text_vote  = SSdbcore.NewQuery("UPDATE [format_table_name("poll_textreply")] SET datetime = Now(), ip = INET_ATON('[client.address]'), replytext = '[replytext]' WHERE pollid = '[pollid]' AND ckey = '[ckey]'")

```

Comment:
As a side thought, this could be done as `INSERT ... DUPLICATE KEY UPDATE` if `pollid-ckey` was a unique key which would save having to actually check if a user has voted.

On the other hand, there's no advantage to that beyond one less proc call and query so can't really say it'd be worth changing.

Reply:
That would probably be best done in another PR, i'd prefer to not touch the schema in this.